File: src/main/java/com/tourlk/entity/TourPackage.java
package com.tourlk.entity;

import com.tourlk.enums.PackageStatus;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.Lob;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;

/**
 * A bookable tour package, owned by the ADMIN or GUIDE who created it.
 * Goes through the DRAFT -&gt; PENDING_APPROVAL -&gt; ACTIVE / INACTIVE
 * lifecycle managed by {@code TourPackageServiceImpl}.
 */
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "tour_packages")
public class TourPackage extends AuditableEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 200)
    private String title;

    @Lob
    @Column(nullable = false)
    private String description;

    /**
     * The place this package visits. Migrated from a free-text String to
     * a real {@link Destination} relation so browsing/filtering by
     * destination is consistent across the platform. Required.
     */
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "destination_id", nullable = false)
    private Destination destination;

    @Column(name = "duration_days", nullable = false)
    private int durationDays;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal price;

    @Column(name = "max_capacity", nullable = false)
    private int maxCapacity;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    @Builder.Default
    private PackageStatus status = PackageStatus.DRAFT;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by_id", nullable = false)
    private User createdBy;

}

File: frontend/src/types/tourPackage.ts
import type { DestinationSummary } from './destination'

export type PackageStatus = 'DRAFT' | 'PENDING_APPROVAL' | 'ACTIVE' | 'INACTIVE' | 'ARCHIVED'

// Matches com.tourlk.dto.TourPackageRequestDto
export interface TourPackageRequestDto {
  title: string
  description: string
  destinationId: number | ''
  durationDays: number
  price: number
  maxCapacity: number
}

// Matches com.tourlk.dto.TourPackageResponseDto
export interface TourPackageResponseDto {
  id: number
  title: string
  description: string
  destination: DestinationSummary
  durationDays: number
  price: number
  maxCapacity: number
  status: PackageStatus
  createdById: number
  createdByName: string
  createdAt: string
}

export interface TourPackageSearchParams {
  destinationId?: number
  minPrice?: number
  maxPrice?: number
}
